### PR TITLE
PPYoloELoss not does not require reg_max parameter

### PR DESCRIPTION
**Pull request by BloodAxe**
This PR drops the necessity to specify `reg_max` argument in the Loss which should match with the reg_max in the model.
We have all the infromation to infer it's value solely from model's outputs.

### CODE REVIEW

File: src/super_gradients/training/losses/ppyolo_loss.py [hunk 1]
```diff
@@ -656,14 +657,19 @@ def __init__(
...
+        :param reg_max:                    (Deprecated) Number of regression bins. Default is None (will be inferred from model's outputs)
         :param use_batched_assignment:     Whether to use batched targets assignment or sequential (per-image).
                                            Default is True (batched).
                                            Batched assignment can be faster when number of the target per image is more or
                                            less the same across the batch, but it has higher peak GPU memory usage.
                                            Sequential assignment has lower peak GPU memory usage and preferable for cases
                                            when number of targets per image varies a lot.
         """
+        if reg_max is not None:
+            warnings.warn(
```
Comment (shaydeci) [inline src/super_gradients/training/losses/ppyolo_loss.py:669]: Please add proper deprecation versions info (deprecated since XXX will be removed in YYY).

**Review by shaydeci — CHANGES_REQUESTED** [review]
Please add deprecation versions, then LGTM on my end.

File: src/super_gradients/training/losses/ppyolo_loss.py (hunk 1 above)
Comment (BloodAxe) [reply src/super_gradients/training/losses/ppyolo_loss.py:669]: Fixed in 2be902acadb623fed44ca7e4b24a31059cfc8ba4

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**